srcCpp/RTIDDSImpl.cxx

```diff
@@ -2724,25 +2708,20 @@ IMessagingReader *RTIDDSImpl<T>::CreateReader(
         }
     }
 
-    if (_transport.transportConfig.kind != TRANSPORT_TCPv4
-            && _transport.transportConfig.kind != TRANSPORT_TLSv4
-            && _transport.transportConfig.kind != TRANSPORT_WANv4
-            && _transport.transportConfig.kind != TRANSPORT_SHMEM
-            && _IsMulticast) {
-
-        const char *multicast_addr;
+    if (_transport.allowsMulticast()) {
+        dr_qos.multicast.value.ensure_length(1, 1);
+        dr_qos.multicast.value[0].receive_address = DDS_String_dup(
+                _transport.getMulticastAddr(topic_name).c_str());
 
-        if (strcmp(topic_name, perftest_cpp::_ThroughputTopicName) == 0) {
-            multicast_addr = THROUGHPUT_MULTICAST_ADDR;
-        } else if (strcmp(topic_name, perftest_cpp::_LatencyTopicName) == 0) {
-            multicast_addr = LATENCY_MULTICAST_ADDR;
-        } else {
-            multicast_addr = ANNOUNCEMENT_MULTICAST_ADDR;
+        if (dr_qos.multicast.value[0].receive_address == NULL) {
+            fprintf(stderr,
+                    "topic name must either be %s or %s or %s.\n",
+                    GetThroughputTopicName().c_str(),
+                    GetLatencyTopicName().c_str(),
+                    GetAnnouncementTopicName().c_str());
+            return NULL;
```

Comment:
Can you verify if it is necessary?

I consider we will not enter here never, because we have verified the topic before.